### PR TITLE
Wonderverse integration

### PR DESCRIPTION
**Pull request by terryli0095**
## Motivation and Context

To pull in wonderverse tasks/contributions and display them in the allocation page on profile summary

## Description

- refactored ContributionUser data structure from array to json/map (address to list of contributions , easier to work with). Now searching for a user's contribution is O(1)
- merge dework and wonder contributions into the same data structure
- add optional `source` type to contribution to denote origin of contributions

## Test and Deployment Plan

thoroughly tested locally, updated the mock data format for contributions

## Screenshots (if appropriate):
https://www.loom.com/share/8ff93944efe34ae7877ef2bb391c8b3f
Screen Shot 2022-09-20 at 1 41 27 PM
Screen Shot 2022-09-20 at 1 41 36 PM
Screen Shot 2022-09-20 at 1 41 52 PM

### CODE REVIEW

**Comment by terryli0095** [comment]
I believe we can close the other pr,

File: src/hooks/useContributions.ts [hunk 1]
```diff
@@ -35,51 +38,84 @@ const mockData: ContributionUser = {
...
+                  .then(res => res as Response);
+              }
+              if (integration.type === 'wonder') {
+                let url = `https://external-api.wonderapp.co/v1/coordinape/contributions?org_id=${integration.data.organizationId}&epoch_start=${epoch.start_date}&epoch_end=${epoch.end_date}`;
+                if (integration.data.podIds) {
+                  for (const podId of integration.data.podIds) {
+                    url += `&pod_ids=${podId}`;
+                  }
+                }
+                console.log('url', url);
```
Comment (crabsinger) [inline src/hooks/useContributions.ts:75]: @terryli0095 lets remove this console's before merge in production

File: src/hooks/useContributions.ts [hunk 2]
```diff
@@ -35,51 +38,84 @@ const mockData: ContributionUser = {
   ],
 };
 
-export function useContributionUsers(): ContributionUser[] {
+export function useContributionUsers(): UserContributions {
   const integrations = useCurrentCircleIntegrations();
   const epoch = useSelectedCircle().circleEpochsStatus.currentEpoch;
-
+  console.log('integrations', integrations);
```
Comment (crabsinger) [inline src/hooks/useContributions.ts:44]: here as well

**Comment by crabsinger** [comment]
@terryli0095 please rebase and remove console logs, then should be good to go!

File: src/hooks/useContributions.ts [hunk 3]
```diff
@@ -35,51 +38,82 @@ const mockData: ContributionUser = {
...
+                let url = `https://external-api.wonderapp.co/v1/coordinape/contributions?org_id=${integration.data.organizationId}&epoch_start=${epoch.start_date}&epoch_end=${epoch.end_date}`;
+                if (integration.data.podIds) {
+                  for (const podId of integration.data.podIds) {
+                    url += `&pod_ids=${podId}`;
+                  }
+                }
+                return fetch(url)
+                  .then(res => res.json())
+                  .then(res => res as Response)
+                  .then(res => res);
```
Comment (crabsinger) [inline src/hooks/useContributions.ts:77]:
```suggestion
```

@terryli0095 do we need line 77? Seems it does't do anything...

Comment (terryli0095) [reply src/hooks/useContributions.ts:77]: fixed

**Review by crabsinger — APPROVED** [review]
Awesome! This looks good and ready for merge. The one remaining thing is the formatting for the epoch dates and the error I experienced, but we can revisit that if it comes up again.

**Review by crabsinger — CHANGES_REQUESTED** [review]
URL still seems broken.

**Review by crabsinger — APPROVED** [review]

**Pull request merged**